### PR TITLE
task1 arm64 asm completed

### PR DESCRIPTION
**Pull request by yyhuang95**
arm64 asm task completed

### CODE REVIEW

File: .github/workflows/c-cpp.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: C/C++ CI
+
+on:
+  push:
+    branches: [ "main" ]
```
Comment (KKiiim) [inline .github/workflows/c-cpp.yml:5]: should ON pull_request that will trigger github action.

File: googleTest.cc [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+//   // Expect two strings not to be equal.
+
+//   EXPECT_EQ(myAbs(-8), 8);
+//   EXPECT_EQ(myAbs(-4294967296), 4294967296);
+//   EXPECT_EQ(myAbs(8), 8);
+//   // Expect equality.
+//   EXPECT_EQ(myAbs(0), 0);
+//   EXPECT_EQ(myAbs(4294967296), 4294967296);
+// }
+
```
Comment (KKiiim) [inline googleTest.cc:17]: Do not need comment other TESTs, gtest supports multi tests that will show the all results in github action .

File: googleTest.cc [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+// TEST(myMemset, BasicAssertions) {
+//     void *buffer = malloc(1024*1024*32);
+//     EXPECT_EQ(myMemset(buffer, 8, 0), 8);
+//     EXPECT_PRED3(AssertMemset, buffer, 8, 0);
+//     EXPECT_EQ(myMemset(buffer, 1024*1024*32, 0), 1024*1024*32);
+//     EXPECT_PRED3(AssertMemset, buffer, 1024*1024*32, 0);
+//     free(buffer);
+// }
+
```
Comment (KKiiim) [inline googleTest.cc:37]: the same.

File: googleTest.cc [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+void QuickSort(int32_t *array, int begin, int end) {
+    if (begin >= end) {
+        return;
+    }
+    int32_t pivot = partition(array, begin, end);
+    printf("pivot=%d,\t", pivot);
+    PrintArray2(array, begin, end);
+    QuickSort(array, begin, pivot - 1);
+    QuickSort(array, pivot + 1, end);
+}
```
Comment (KKiiim) [inline googleTest.cc:74]: All qsort function details should implement with arm64 assembly. In test, the caller only call single function and get result. That means, the function QuickSort logic should also assembly written.

**Review by KKiiim — CHANGES_REQUESTED** [review]

**Comment by yyhuang95** [comment]
branch has moved from my own repo to arm64 branch in this repo, I will fix these problems and open a new pull-request

**Pull request closed without merging**